File: app/src/main/java/com/vsa/filmoteca/data/usecase/GetMovieDetailUseCase.java
package com.vsa.filmoteca.data.usecase;

import com.vsa.filmoteca.data.repository.MoviesDataRepository;

import rx.Observable;

/**
 * Created by albertovecinasanchez on 18/7/16.
 */

public class GetMovieDetailUseCase {

    private MoviesDataRepository mMoviesDataRepository;

    public GetMovieDetailUseCase(MoviesDataRepository mMoviesDataRepository) {
        this.mMoviesDataRepository = mMoviesDataRepository;
    }

    public Observable<String> movieDetail(String url) {
        return mMoviesDataRepository.movieDetail(url);
    }

}

File: app/src/main/java/com/vsa/filmoteca/domain/usecase/GetMovieDetailUseCase.kt
package com.vsa.filmoteca.domain.usecase

import com.vsa.filmoteca.data.source.repository.MoviesDataRepository
import com.vsa.filmoteca.network.executor.AsyncExecutor
import javax.inject.Inject

/**
 * Created by albertovecinasanchez on 18/7/16.
 */

class GetMovieDetailUseCase @Inject constructor(
    private val repository: MoviesDataRepository,
    private val asyncExecutor: AsyncExecutor
) {

    fun movieDetail(url: String, callback: (Result<String>) -> Unit) {
        asyncExecutor.execute({
            repository.movieDetail(url)
        }, callback)
    }

}
